cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -1,19 +1,79 @@
 package kubecontroller
 
 import (
+	"fmt"
+	"net"
+
 	"github.com/Azure/azure-container-networking/cns"
 	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
 )
 
-// CRDStatusToNCRequest translates a crd status to network container request
-func CRDStatusToNCRequest(crdStatus *nnc.NodeNetworkConfigStatus) (*cns.CreateNetworkContainerRequest, error) {
-	//TODO: Translate CRD status into network container request
-	//Mat will pick up from here
-	return nil, nil
+// CRDStatusToNCRequest translates a crd status to createnetworkcontainer request
+func CRDStatusToNCRequest(crdStatus nnc.NodeNetworkConfigStatus) (*cns.CreateNetworkContainerRequest, error) {
+	var (
+		ncRequest         cns.CreateNetworkContainerRequest
+		nc                nnc.NetworkContainer
+		secondaryIPConfig cns.SecondaryIPConfig
+		ipSubnet          cns.IPSubnet
+		ipAssignment      nnc.IPAssignment
+		err               error
+		ip                net.IP
+		ipNet             *net.IPNet
+		bits              int
+	)
+
+	ncRequest.SecondaryIPConfigs = make(map[string]cns.SecondaryIPConfig)
+
+	for _, nc = range crdStatus.NetworkContainers {
```

Comment:
We should check if crdStatus has any NetworkContainers, if there are none then return nil, else pick the first one (for now we have supported only single nic) and populate the request. We can also throw error if range has > 1 NC

Reply:
Added a check that length of NetworkContainers list is 1